lewis/adapters/stream.py

```diff
@@ -163,8 +186,12 @@ def __init__(self, func, pattern, argument_mappings=None, return_mapping=None, d
             raise RuntimeError('Can not construct a Func-object from a non callable object.')
 
         self.func = func
-        self.raw_pattern = pattern
-        self.pattern = re.compile(b(pattern), 0) if pattern else None
+
+        self.raw_pattern = pattern.raw_pattern
+        self.pattern = re.compile(b(pattern.regex_pattern), 0)
```

Comment:
I see your point that it is more convenient to define the patterns with less stuff to type and also regarding the implementation complexity. But there are still some things I'm a bit worried about with this approach in decreasing order of "worriedness":

1. Breaking existing patterns. Regular expressions that people have written so far which do not contain a `^` at the beginning will stop working. I don't know how many there are, but it may cause some surprises. It can be documented in the release notes and now we also learned about upgrade guides, but what I find dangerous in this case is that there's no easy way to warn users when there is no `^` at the beginning of the pattern - it might be intended or not, there's no way to tell.
2. "With great convenience comes some danger"...I'm a bit afraid that one character may be easily overlooked or forgotten. That is also true for the `pattern_type` approach to some extent, but if nothing is specified, all strings will be interpreted as regular expressions consistently.
3. Extensibility. Although this may not be too likely to happen in the nearest future, if it decides to happens anyway, it is quite likely that the implementation needs to change (or am I overlooking something?). And then I fear it might be more difficult to come up with a way of doing that while preserving the old behavior (which is the new behavior we're trying to come up with now).

I do find the "not typing stuff" tempting, but as explained above, I think it might be the slightly more dangerous route to go (even setting aside point 3, which is probably too unlikely to happen anytime soon). I'd rather accept the complexity on our side and explain the `pattern_type` to the user - it's only there this feature really needs explaining besides the `fmt`/`regex`-classes. The other changes are internal and only concern us.

Is there a third way that we're not seeing?